File: authorization-admin-service-spring/src/main/java/eu/rudisch/authorization/admin/config/RestTokenKeyFetcher.java
package eu.rudisch.authorization.admin.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

import eu.rudisch.authorization.admin.web.controller.resource.TokenKey;

@Component
public class RestTokenKeyFetcher {

	@Value("${custom.oauth.server}")
	private String oauthServer;

	String getTokenKey() {
		final RestTemplate restTemplate = new RestTemplate();
		TokenKey tokenKey = restTemplate.getForObject(oauthServer + "/oauth/token_key", TokenKey.class);
		return tokenKey != null ? tokenKey.getValue() : "";
	}

}

File: authorization-admin-service-spring-kotlin/src/main/kotlin/eu/rudisch/authorization/admin/config/RestTokenKeyFetcher.kt
package eu.rudisch.authorization.admin.config

import eu.rudisch.authorization.admin.web.controller.resource.TokenKey
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import org.springframework.web.client.RestTemplate

@Component
class RestTokenKeyFetcher {
	@Value("\${custom.oauth.server}")
	private val oauthServer: String? = null

	internal fun getTokenKey(): String {
		val restTemplate = RestTemplate()
		val tokenKey = restTemplate.getForObject<TokenKey>("$oauthServer/oauth/token_key", TokenKey::class.java)
		return tokenKey?.value ?: ""
	}
}